recipes/CVSS/S2ST/train_w2v2.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env python3
+
+import sys
+import json
+import itertools
+import torch
+import copy
+import pathlib as pl
+from hyperpyyaml import load_hyperpyyaml
+import speechbrain as sb
+from speechbrain.utils.distributed import run_on_main
+from speechbrain.utils.data_utils import scalarize
+import torch
+import torchaudio
+import os
+import numpy as np
+import random
+
+from utils.embedding import EmbeddingManager
+from torch.nn import DataParallel
+from torch.nn.parallel import DistributedDataParallel
+
+#torch.autograd.set_detect_anomaly(True)
+
+
+# Define training procedure
+class S2U(sb.core.Brain):
+    def compute_forward(self, batch, stage):
+        """Forward computations from the waveform batches to the output probabilities."""
+        batch = batch.to(self.device)
+        wavs, wav_lens = batch.sig  # audio
+        tokens_bos, _ = batch.unit_bos  # speech unit _bos
+
+        # replace pad_index by 0 in the signal
+        wavs[wavs == self.hparams.pad_index] = 0.0
+
+        # compute features
+        feats = self.modules.wav2vec2(wavs, wav_lens)
+
+        # dimensionality reduction
+        src = self.modules.enc(feats)
+
+        # Patch to allow DP or DDP training
+        if isinstance(self.modules.transformer, DataParallel):
```

Comment:
remove DataParallel case, we aim at not supporting it anymore. What you could do, since they have identical process (DDP an DP) could be to add the multi GPU support as an else statement to support both without explicitly writing that we support DP.